server/data/user.go

```diff
@@ -132,3 +141,40 @@ func IsValidLogin(username string, password string) bool {
 
 	return true
 }
+
+func SearchUsers(field string, query string) []*UserProfile {
```

Comment:
This file is getting pretty big. Maybe we could move this function to a file called `data/search.go`, along with the `searchFields` map?